### PR TITLE
Convert to a Python package

### PR DESCRIPTION
**Pull request by khaeru**
Excepting tests and helper code for us devs, the package consists of two lines that define one object:
```
from iam_units import registry
```

How to review:
- Will merging this PR break any existing code? If not, then let's handle any adjustments in additional PRs to keep things quick and tidy.
- The name 'units' is taken on PyPI: https://pypi.org/project/units/, so I went with 'iam_units'. Good? I could also live with 'ia_units'.

### CODE REVIEW

**Comment by gidden** [comment]
@khaeru took a look and all seems very immaculate. @znicholls, FYI that this is now installable. @danielhuppmann I guess we can do a quick patch on pyam to depend on this?

Also, very happy to see that `setup.py` now looks like it's actually easy to implement instead of the complicated nonsense I've had to work with for the past 10 years!

**Comment by gidden** [comment]
Ah, I forgot to mention. We use `pyam-iamc` name on PyPI since `pyam` was taken. That could be one pattern to follow here.

File: iam_units/update.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+from pathlib import Path
+
+import pandas as pd
+import sys
+
+
+DATA_PATH = Path(__file__).parent / 'data'
+
+# list of contexts
+contexts = ['SARGWP100', 'AR4GWP100', 'AR5GWP100']
```
Comment (danielhuppmann) [inline iam_units/update.py:10]: suggest to rename to `gwp_contexts` assuming that we'll want to add other context-types in the future (e.g., currency conversion)

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[metadata]
+name = iam_units
+author = Integrated Assessment Modeling Consortium members
```
Comment (danielhuppmann) [inline setup.cfg:3]: suggest to change to actual authors (see also my comment on copyright notice)

**Comment by danielhuppmann** [comment]
> @danielhuppmann I guess we can do a quick patch on pyam to depend on this?

Yes, @dependabot proves quite useful here. But unless this is also released via conda and/or pip, I don't think we'll get around the submodule hassle (but that's a minor hassle now that it's set up).

**Comment by danielhuppmann** [comment]
> We use `pyam-iamc` name on PyPI since `pyam` was taken. That could be one pattern to follow here.

-1. Don't think that `units-iamc` has a nice ring to it...

**Comment by znicholls** [comment]
> @znicholls, FYI that this is now installable

Thanks @gidden  and thanks @danielhuppmann for your comments about authorship. Both appreciated and this looks like a very easy maintenance way of getting the behaviour you need.

For completeness, we needed a units repository in silicone which had wide gas and gwp context coverage so I've actually already split out https://github.com/openscm/openscm-units from https://github.com/openscm/scmdata. I didn't raise this here as I thought the discussions in #9 suggested a package wasn't desired. I'm happy to work out a way to combine openscm-units and this repo into a shared resource if there's desire. Also fine to keep them separate for the time being though (openscm-units certainly doesn't have the feature of being 'as low-maintenance as possible' because of its desire to include more features).

> But unless this is also released via conda and/or pip, I don't think we'll get around the submodule hassle

+1 from me, you can't release a package on PyPI that has url based install dependencies. If you want an automated deployment workflow, the jobs from [here](https://github.com/openscm/openscm-units/blob/007066de63f1bed73842f4b3e53867c0d1c6f4b5/.github/workflows/ci-cd-workflow.yml#L108) onwards might be helpful (you need to set the right secrets, see [details here](https://github.com/pypa/gh-action-pypi-publish#usage) and alter hard-coded names but then you're done).

File: iam_units/update.py (hunk 1 above)
Comment (khaeru) [reply iam_units/update.py:10]: Will move it inside the function, should have done that in the first place.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
+Unit definitions for integrated-assessment research
+***************************************************
+
+© 2020 `IAMC members`_; licensed under the `Creative Commons Attribution 4.0 license`_.
```
[the remark this answers is not in the document]
Comment (khaeru) [reply README.rst:4]: Okay, will go with an AUTHORS file

**Comment by khaeru** [comment]
Sorry @danielhuppmann @gidden —one additional point based on https://github.com/IAMconsortium/units/pull/16#discussion_r403705482: since this is now no longer simply content (a text file) but code, I will change the license to GPL3. Good?

@znicholls thanks for the pointer to the GitHub workflow examples and information about the other package.

**Comment by danielhuppmann** [comment]
> Sorry @danielhuppmann @gidden —one additional point based on [#16 (comment)](https://github.com/IAMconsortium/units/pull/16#discussion_r403705482): since this is now no longer simply content (a text file) but code, I will change the license to GPL3. Good?

Fine for me!

File: AUTHORS [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Daniel Huppmann <huppmann@iiasa.ac.at> (@danielhuppmann, IIASA)
+Franceso Lovat <lovat@iiasa.ac.at> (@francescolovat, IIASA)
+Paul Natsuo Kishimoto <kishimot@iiasa.ac.at> (@khaeru, IIASA)
```
Comment (danielhuppmann) [inline AUTHORS:3]: I would feel it's only fair if @khaeru is the first name on the list of authors...

```suggestion
Paul Natsuo Kishimoto <kishimot@iiasa.ac.at> (@khaeru, IIASA)
Daniel Huppmann <huppmann@iiasa.ac.at> (@danielhuppmann, IIASA)
Franceso Lovat <lovat@iiasa.ac.at> (@francescolovat, IIASA)
```

Comment (khaeru) [reply AUTHORS:3]: Thanks!

**Pull request merged**